src/main/java/org/opensearch/index/codec/customcodecs/backward_codecs/lucene99/QatCompressionMode.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec.customcodecs.backward_codecs.lucene99;
+
+import org.apache.lucene.codecs.compressing.CompressionMode;
+import org.apache.lucene.codecs.compressing.Compressor;
+import org.apache.lucene.codecs.compressing.Decompressor;
+import org.apache.lucene.store.ByteBuffersDataInput;
+import org.apache.lucene.store.DataInput;
+import org.apache.lucene.store.DataOutput;
+import org.apache.lucene.util.ArrayUtil;
+import org.apache.lucene.util.BytesRef;
+import org.opensearch.index.codec.customcodecs.QatZipperFactory;
+
+import java.io.IOException;
+import java.util.function.Supplier;
+
+import com.intel.qat.QatZipper;
+
+/** QatCompressionMode offers QAT_LZ4 and QAT_DEFLATE compressors. */
+public class QatCompressionMode extends CompressionMode {
+
+    private static final int NUM_SUB_BLOCKS = 10;
+
+    private final QatZipper.Algorithm algorithm;
+    private final int compressionLevel;
+    private final Supplier<QatZipper.Mode> supplier;
+
+    /** default constructor */
+    protected QatCompressionMode() {
+        this(Lucene99QatCodec.DEFAULT_COMPRESSION_MODE, Lucene99QatCodec.DEFAULT_COMPRESSION_LEVEL, () -> {
+            return Lucene99QatCodec.DEFAULT_QAT_MODE;
+        });
+    }
+
+    /**
+     * Creates a new instance.
+     *
+     * @param mode The compression mode (QAT_LZ4 or QAT_DEFLATE)
+     */
+    protected QatCompressionMode(Lucene99QatCodec.Mode mode) {
+        this(mode, Lucene99QatCodec.DEFAULT_COMPRESSION_LEVEL, () -> { return Lucene99QatCodec.DEFAULT_QAT_MODE; });
+    }
+
+    /**
+     * Creates a new instance.
+     *
+     * @param mode The compression mode (QAT_LZ4 or QAT_DEFLATE)
+     * @param compressionLevel The compression level to use.
+     */
+    protected QatCompressionMode(Lucene99QatCodec.Mode mode, int compressionLevel) {
+        this(mode, compressionLevel, () -> { return Lucene99QatCodec.DEFAULT_QAT_MODE; });
+    }
+
+    /**
+     * Creates a new instance.
+     *
+     * @param mode The compression mode (QAT_LZ4 or QAT_DEFLATE)
+     * @param compressionLevel The compression level to use.
+     * @param supplier a supplier for QAT acceleration mode.
+     */
+    protected QatCompressionMode(Lucene99QatCodec.Mode mode, int compressionLevel, Supplier<QatZipper.Mode> supplier) {
```

Comment:
I don't think we should duplicate QATCompressionMode. This class should be reusable right? We do it for Zstd related CompressionModes currently.